src/components/recordedit/multi-form-input-row.tsx

```diff
@@ -257,10 +263,79 @@ const MultiFormInputRow = ({
     setShowApplyAllSpinner(false);
   };
 
-  // -------------------------- render logic ---------------------- //
+  /**
+   * if the fk input has a domain-filter, we have to make sure it's identical
+   * for all the selected rows. otherwise we won't be able to offer this feature.
+   */
+  const checkForeignKeyDomainFilter = async () => {
+    let domainFilterFormNumber = activeForms && activeForms.length > 0 ? activeForms[0] : 1;
 
-  const colName = cm.column.name;
-  const inputName = `${MULTI_FORM_INPUT_FORM_VALUE}-${colName}`;
+    if (!cm.hasDomainFilter) {
+      return { allowed: true, domainFilterFormNumber };
+    }
+
+    // remove the existing error
+    clearErrors(inputName);
+
+    if (!activeForms || activeForms.length < 2) {
+      return { allowed: true, domainFilterFormNumber }
+    }
+
+    // make sure the computed references are the same
+    let computedRefURI = '';
+    const areAllSame = activeForms.every((formNumber: number, formIndex: number) => {
+      const ref = createForeignKeyReference(cm.column, reference, formNumber, foreignKeyData, getValues);
+      if (formIndex === 0) {
+        domainFilterFormNumber = formNumber;
+        computedRefURI = ref.uri;
+      } else {
+        return computedRefURI === ref.uri;
+      }
+      return true;
```

Comment:
I think this would read cleaner as:
```
if (formIndex === 0) {
  domainFilterFormNumber = formNumber;
  computedRefURI = ref.uri;
  return true
}

return computedRefURI === ref.uri;
```